### PR TITLE
Add context_window_size support for LLM

### PR DESCRIPTION
**Pull request by gaya3-vijayakumar**
## What

Add context_window_size support for LLM

## Why

To  indicate to user about what context size the LLM can support

## How

...

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions / Env Variables

-

## Notes on Testing

...

## Screenshots

...

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/adapters/llm/azure_open_ai/src/azure_open_ai.py [hunk 1]
```diff
@@ -75,3 +73,7 @@ def test_connection(self) -> bool:
         llm = self.get_llm_instance()
         test_result: bool = LLMHelper.test_llm_instance(llm=llm)
         return test_result
+
+    def get_context_window_size(self) -> int:
```
Comment (muhammad-ali-e) [inline src/unstract/adapters/llm/azure_open_ai/src/azure_open_ai.py]: Is it allways return zero?

File: src/unstract/adapters/llm/llm_adapter.py [hunk 2]
```diff
@@ -48,3 +49,16 @@ def get_llm_instance(self) -> Optional[LLM]:
...
+    def get_context_window_size(self) -> int:
+        """Get the context window size supported by the LLM.
+
+        Returns:
+            int: Context window size supported by the LLM
+        """
+        context_window_size: int = 0
+        llm = self.get_llm_instance()
+        if llm is not None:
+            context_window_size = llm.metadata.context_window
```
Comment (muhammad-ali-e) [inline src/unstract/adapters/llm/llm_adapter.py:64]: What is the default context_window if we are not set, Is it None?
Is the llm.metadata always conatins context_window ?

File: src/unstract/adapters/llm/azure_open_ai/src/azure_open_ai.py (hunk 1 above)
Comment (gaya3-vijayakumar) [reply src/unstract/adapters/llm/azure_open_ai/src/azure_open_ai.py]: Sorry. Guess this is a wrong check-in. Removed this.

Comment (gaya3-vijayakumar) [reply src/unstract/adapters/llm/azure_open_ai/src/azure_open_ai.py]: Thank you for catching this @muhammad-ali-e

File: src/unstract/adapters/llm/llm_adapter.py (hunk 2 above)
Comment (gaya3-vijayakumar) [reply src/unstract/adapters/llm/llm_adapter.py:64]: context_window is not stored anywhere for now. It is dynamically fetched only while constructing the response for the Get adapter details API.

Comment (gaya3-vijayakumar) [reply src/unstract/adapters/llm/llm_adapter.py:64]: I initialized it here so that we do not get a run time error in case llm is null for some odd reason.

**Review by athul-rs — APPROVED** [review]

**Review by muhammad-ali-e — APPROVED** [review]
LGTM

File: src/unstract/adapters/llm/azure_open_ai/src/azure_open_ai.py [hunk 3]
```diff
@@ -39,10 +40,7 @@ def get_description() -> str:
 
     @staticmethod
     def get_icon() -> str:
-        return (
-            "/icons/"
-            "adapter-icons/AzureopenAI.png"
-        )
+        return "/icons/" "adapter-icons/AzureopenAI.png"
```
Comment (chandrasekharan-zipstack) [inline src/unstract/adapters/llm/azure_open_ai/src/azure_open_ai.py]:
```suggestion
        return "/icons/adapter-icons/AzureopenAI.png"
```

File: src/unstract/adapters/llm/llm_adapter.py (hunk 2 above)
Comment (chandrasekharan-zipstack) [reply src/unstract/adapters/llm/llm_adapter.py:64]: @gaya3-zipstack why do we return `Optional[LLM]` in `get_llm_instance()`? This leads to an extra check at all places of use - and in most scenarios we raise an exception. Instead can we always raise an exception if no valid LLM is found.
Additionally we can control the behavior of raising an exception with a flag

Comment (gaya3-vijayakumar) [reply src/unstract/adapters/llm/llm_adapter.py:64]: @chandrasekharan-zipstack yeah we could do that. That will be a good improvement. Needs to be done in the SDK also. Can we decouple it with this PR?

**Review by chandrasekharan-zipstack — APPROVED** [review]

**Pull request merged**